### PR TITLE
Gh action tag image

### PR DESCRIPTION
**Pull request by johrstrom**
Gh action tag image and push it to the OSC container registry.

### CODE REVIEW

File: .github/workflows/publish_image.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    name: Push Docker image to OSC Container Registry
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+      - name: setup the environment
+        run: |
+          TAG=${GITHUB_REF#refs/*/}
+          REPO="docker-registry.osc.edu/gateways"
+          echo "LATEST_IMG=$REPO/phylogatr:latest" >> $GITHUB_ENV
```
Comment (treydock) [inline .github/workflows/publish_image.yml:23]: I have a vague recollection that this is rejected by Github actions due to security.  This is how you save values:

```
echo ::set-output name=latest::$LATEST_IMG
echo ::set-output name=version::$VERSIONED_IMG
```

Then you reference like this:

```
${{ steps.setup.outputs.latest }}
${{ steps.setup.outputs.version }}
```

The step doing setup would need this:

```
id: setup
```

I think it's a lot simpler to just setup output for tag version and reference that and do like 

```
docker push docker-registry.osc.edu/gateways/phylogatr:${{ steps.setup.outputs.version }}
docker push docker-registry.osc.edu/gateways/phylogatr:latest
```

**Comment by treydock** [comment]
So there are actions to handle some of this and they are published by Docker developers so seems legit:

```
      - name: Login to DockerHub
        uses: docker/login-action@v1 
        with:
          username: ${{ secrets.DOCKER_HUB_USERNAME }}
          password: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
```

There is also one to build and push:

```
      - name: Build and push
        id: docker_build
        uses: docker/build-push-action@v2
        with:
          context: ./
          file: ./Dockerfile
          push: true
          tags: ohiosupercomputer/job-pod-reaper:${{ steps.version.outputs.version }}
```

Can do multiple tags if you want

File: .github/workflows/publish_image.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+name: Publish Docker image
+
+on:
+  push:
+    tags:
+      - '*'
```
Comment (treydock) [inline .github/workflows/publish_image.yml:6]: Would it be beneficial to run on pull requests and tags and then limit the push to just tags?  I think if you did that might be clearer and simpler to put the pull request build into separate file.

Comment (johrstrom) [reply .github/workflows/publish_image.yml:6]: May as well I guess.

**Comment by johrstrom** [comment]
Looks like we need Gitlab to connect to internal OSC repos

https://github.com/OSC/phylogatr-web/blob/a0df1de94db3350fe91f812af1c97630fc4f3075/Dockerfile#L48-L49

**Comment by treydock** [comment]
@johrstrom The only repo OSC would need is SLURM repo and rather than use SLURM repo could just build from source.  If SLURM is executed in the container then don't need Lua plugin, that's only needed for how I build RPMs for OSC. I'd recommend to just build SLURM in the container from source using version we use at OSC which is 20.11 release series.

**Comment by treydock** [comment]
Or you can install from EPEL, SLURM is in EPEL now but they do major version updates so sometime in future they may have 21.08 in EPEL but right now it's still 20.11.8 which is compatible with our systems.

**Comment by johrstrom** [comment]
Thanks. I already got the image to build in gitlab, so I guess we'll just keep it there.

**Pull request closed without merging**